modules/mod-aup/ImportAUP.cpp

```diff
@@ -395,6 +398,72 @@ void AUPImportFileHandle::Import(ImportProgressListener& progressListener,
    for (auto pClip : mClips)
       pClip->UpdateEnvelopeTrackLen();
 
+   
+   Track* unlinkedTrack {};
+   for(const auto& t : tracks)
+   {
+      const auto linkType = t->GetLinkType();
+      // Note, the next function may have an important upgrading side effect,
+      // and return no error; or it may find a real error and repair it, but
+      // that repaired track won't be used because opening will fail.
+      if (!t->LinkConsistencyFix()) {
+         SetError(XO("A channel of a stereo track was missing."));
+         break;
+      }
+      if(unlinkedTrack != nullptr)
+      {
+         //Not an elegant way to deal with stereo wave track linking
+         //compatibility between versions
+         if(const auto left = dynamic_cast<WaveTrack*>(unlinkedTrack))
+         {
+            if(const auto right = dynamic_cast<WaveTrack*>(t))
+            {
+               left->SetPan(-1.0f);
+               right->SetPan(1.0f);
+               if(mErrorMsg.empty())
+               {
+                  SetWarning(XO(
+//i18n-hint: Text of the message dialog that may appear on attempt
+//to import an AUP project.
+//%s will be replaced with an explanation of the actual reason of
+//project modification.
+"%s\n"
+"This feature is not supported in Audacity versions past 3.3.3.\n"
+"These stereo tracks have been split into mono tracks.\n"
+"Please verify that everything works as intended before saving.")
+                     .Format(WaveTrackUtils::GetChannelUnlinkReason(*left, *right))
+                  );
+               }
+            }
+         }
+         unlinkedTrack = nullptr;
+      }
+
+      if(linkType != ChannelGroup::LinkType::None &&
+         t->GetLinkType() == ChannelGroup::LinkType::None)
+      {
+         //Wait when LinkConsistencyFix is called on the second track
+         unlinkedTrack = t;
+      }
+
+      if (const auto message = t->GetErrorOpening())
+      {
+         SetError(*message);
+         break;
+      }
+   }
```

Comment:
This looks correct but it's all duplication of a part of `ProjectFileManager::ReadProjectFile`.  Can there be just one function to do this?